src/main/java/com/p6spy/engine/spy/P6DataSource.java

```diff
@@ -301,6 +301,9 @@ public Connection getConnection() throws SQLException {
     try {
       conn = ((DataSource) realDataSource).getConnection();
       connectionInformation.setConnection(conn);
+
+      connectionInformation.setUrl(conn.getMetaData().getURL());
```

Comment:
please add null check: `if (conn.getMetaData() != null)`, as this is causing our test to fail https://api.travis-ci.org/v3/job/454234123/log.txt